### PR TITLE
Guard for null site url when viewing site

### LINKED ISSUE #6586: Crash: NPE due to null url string in ActivityLauncher.openUrlExternal
```
Fatal Exception: java.lang.NullPointerException: uriString
       at android.net.Uri$StringUri.<init>(Uri.java:476)
       at android.net.Uri$StringUri.<init>(Unknown Source)
       at android.net.Uri.parse(Uri.java:438)
       at org.wordpress.android.ui.ActivityLauncher.openUrlExternal(ActivityLauncher.java:330)
       at org.wordpress.android.ui.ActivityLauncher.viewCurrentSite(ActivityLauncher.java:154)
       at org.wordpress.android.ui.main.MySiteFragment$2.onClick(MySiteFragment.java:162)
       at android.view.View.performClick(View.java:6256)
       at android.view.View$PerformClick.run(View.java:24697)
       at android.os.Handler.handleCallback(Handler.java:789)
       at android.os.Handler.dispatchMessage(Handler.java:98)
       at android.os.Looper.loop(Looper.java:164)
       at android.app.ActivityThread.main(ActivityThread.java:6541)
       at java.lang.reflect.Method.invoke(Method.java)
       at com.android.internal.os.Zygote$MethodAndArgsCaller.run(Zygote.java:240)
       at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:767)
```

### PR DESCRIPTION
**Pull request by hypest**
Fixes #6586 

There are cases where the site URL can be null in the model so, need to guard for that when trying to open the site (via MySite).

To test:
* No steps known

### CODE REVIEW

File: WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java [hunk 1]
```diff
@@ -162,7 +162,14 @@ public static void viewCurrentSite(Context context, SiteModel site, boolean open
         AnalyticsTracker.Stat stat = openFromHeader ? AnalyticsTracker.Stat.OPENED_VIEW_SITE_FROM_HEADER
                 : AnalyticsTracker.Stat.OPENED_VIEW_SITE;
         AnalyticsUtils.trackWithSiteDetails(stat, site);
-        openUrlExternal(context, site.getUrl());
+
+        final String siteUrl = site.getUrl();
```
Comment (nbradbury) [inline WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Given the rarity of this bug, what do you think about doing away with the additional string resource and changing the check at the top of this method from `if (site == null)`  to `if (site == null || site.getUrl() == null)`?

Comment (hypest) [reply WordPress/src/main/java/org/wordpress/android/ui/ActivityLauncher.java]: Fair point @nbradbury . I refactored the null checks a bit with 891c6d3. BTW, the Track now fires even on null SiteModel so we can have good Analytics of the user trying to view the site.

I removed the differentiated toast _to the user_ as you suggested. I kept the log message though as it's useful for debugging (think: when user contacts Support).

**Comment by hypest** [comment]
Ready for another pass @nbradbury ! Thanks!

**Comment by nbradbury** [comment]
:shipit:

**Pull request merged**